int64 target_num
---
float64 x
float64 y
